lib/dockerfile-parser.ts

```diff
@@ -0,0 +1,114 @@
+import { CommandEntry, parse } from 'docker-file-parser';
+
+const LiveCommandDirective = 'dev-cmd-live';
+const EscapeDirective = 'escape';
+
+export function parseDockerfile(content: string | Buffer): CommandEntry[] {
+	// This function may look a little weird, but due to bugs
+	// in the comment parsing of docker-file-parser we first go
+	// through the dockerfile, find out directives, and keep a
+	// track of the line number that it was encountered at.
+	// We then pass to the docker-file-parser parsing
+	// function, with no comments. We then integrate the
+	// directive into the returned values using the line
+	// number (as the order matters).
+
+	// issue with comments in docker-file-parser:
+	// https://github.com/joyent/node-docker-file-parser/issues/8
```

Comment:
The last commit in node-docker-file-parser was 2 years ago, I'm not sure anyone will look at this issue.
Maybe we should fork it and fix it ?
Or translate https://github.com/moby/buildkit/blob/master/frontend/dockerfile/parser/parser.go to javascript ?

Reply:
Yeah I've definitely considered this, and started to do some investigation into it. It's more of a time issue than thinking it's not worth it really.